### ISSUE TITLE
Implement search API with an empty response

### ISSUE THREAD

**Issue by thewilkybarkid** [libero/libero#94]
Implements https://github.com/libero/libero/issues/92 without returning any data.

- [ ] Determine framework
- [ ] Create search project
- [ ] Implement search endpoint

**Comment by GiancarloFusiello** [comment]
@thewilkybarkid @giorgiosironi Can I start work on this? Regarding the framework, I plan on using python Flask. Any objections?

**Comment by thewilkybarkid** [comment]
Flask's fine by me.

Ideally we want all the code to be in libraries (so the application itself is just glue/config), so if someone wants to make substantive changes they don't have to fork the whole thing. It's fine to leave it in the application for now (too difficult to development), but worth thinking about the separation. In Browser we've split them into libraries but kept them in the same repo ([code in a dedicated folder](https://github.com/libero/browser/tree/master/vendor-extra), [enabling through the dependency manager](https://github.com/libero/browser/blob/dc4c0b98ec6b722821a9e1f8532908ee2a8c2323/composer.json#L59-L64)), so moving them into dedicated libraries in the future should be trivial.

**Comment by GiancarloFusiello** [comment]
Can we add this to the Cambridge board for visibility?

**Comment by giorgiosironi** [comment]
Yes, it should be put there. This is achieved using the URL of the issues and the `+` in the `To do` column, can show in person if needed.

**Comment by GiancarloFusiello** [comment]
I have created repo [libero/search](https://github.com/libero/search) with basic code to get the project started. Please have a look and I can create pull requests for any changes. 

I have only created a `dev` container at the moment. @giorgiosironi shall we go through creating a prod/ci container with nginx and uwsgi?

**Comment by giorgiosironi** [comment]
The current [`Dockerfile`](https://github.com/libero/search/blob/master/docker/Dockerfile.dev) doesn't look much different from an application container to deploy, apart from the lack of a `CMD` and possibly `uwsgi` as a dependency.

> creating a prod/ci container with nginx and uwsgi?

A container is supposed to have only one top level process, so it would be running `uwsgi`. Though it doesn't strictly need to be `uwsgi`, it could be any application server that exposes the WSGI protocol on a port; `nginx` then has official images that we can use.

I guess for local development you won't have an nginx because the Flask development server exposes directly HTTP instead.

By the way, we'll cover all of this in https://github.com/libero/libero/issues/96 which should define a production-oriented image, publish it, and use it.

**Comment by giorgiosironi** [comment]
> Please have a look and I can create pull requests for any changes.

Can't effectively provide comments situated in the code due to the first commit being already in `master`, we usually try to put in an empty `README` as the first commit and then open the first PR. Now, disparate comments here:
- https://github.com/libero/search/blob/master/tests/test_search.py#L8 is there a way to assert with an integer rather than with a string? We'll have lots of these tests and the status line `Not Found|Permanent Redirect|...` should be unnecessary
- https://github.com/libero/search/blob/master/tests/test_search.py#L9 like for JSON responses should parse the body to check there are no elements in `>item-list>` rather then asserting on the string, as string equality is not robust to white space and formatting
- https://github.com/libero/search/blob/master/tests/test_search.py#L11 I do think it's a good `Content-Type` for now, there may be a more specific one but I'm not an expert
- could specify a `Cache-Control` header, such as "don't cache anything for now" as we'll get to it later?
- https://github.com/libero/search/blob/master/pyproject.toml#L3 I think we started from `0.0.1` for [other projects](https://github.com/libero/pattern-library/releases)
- https://github.com/libero/search/blob/master/pyproject.toml#L9 specifies Python 3.7, I don't have any problems with that but good to hear from others that are more involved in the day-to-day Python
- might want to copy the [milestone description](https://github.com/libero/libero/milestone/12) somewhere before that milestone gets shut
- https://github.com/libero/search/blob/master/docker/Dockerfile.dev#L8 this is from Continuum projects, but we are not obliged to follow it; any other standard? e.g. `/app`
- https://github.com/libero/search/blob/master/docker/dev.env#L2 we tend to go for `dev` as an environment name, not sure if this affects anything

**Comment by GiancarloFusiello** [comment]
Yes, I see what you're saying. I'll be sure to just commit a README first in future.

**Comment by thewilkybarkid** [comment]
Going to suggest we do reset the repo, since there’s a lot here that needs reviewing. Don’t think you can revert the first commit so will have to recreate the repo, but not a problem as nothing will be lost. A first commit should be something like https://github.com/libero/dummy-api/commit/164653b086d158a018a8cb15f9dc54050d482b69.

**Comment by GiancarloFusiello** [comment]
This is now resolved and the [pull request](https://github.com/libero/search/pull/1) for this task is ready for review.
